Source/InitialConditions/BlackHoles/TwoPuncturesInitialData.impl.hpp

```diff
@@ -0,0 +1,88 @@
+/* GRChombo
+ * Copyright 2012 The GRChombo collaboration.
+ * Please refer to LICENSE in GRChombo's root directory.
+ */
+
+#if !defined(TWOPUNCTURESINITIALDATA_HPP_)
+#error "This file should only be included through TwoPuncturesInitialData.hpp"
+#endif
+
+#ifndef TWOPUNCTURESINITIALDATA_IMPL_HPP_
+#define TWOPUNCTURESINITIALDATA_IMPL_HPP_
+
+void TwoPuncturesInitialData::compute(Cell<double> current_cell) const
+{
+    Vars<double> vars;
+    // Set only the non-zero components explicitly below
+    VarsTools::assign(vars, 0.);
+
+    Coordinates<double> coords(current_cell, m_dx, m_center);
+    Tensor<2, double> h_phys, extrinsic_K;
+    Tensor<1, double> shift, Z3;
+    double lapse, Theta;
+
+    interpolate_tp_vars(coords, h_phys, extrinsic_K, lapse, shift, Theta, Z3);
+
+    // analytically set Bowen-York properties below (e.g. conformal flatness,
+    // tracefree K)
+
+    // metric variables
+    vars.chi = 1.0 / h_phys[0][0];
```

Comment:
I know this works for conformal data but I'd be happier if it calculated the determinant explicitly and set it via that for readability (and in case anyone copies the code). In the same way I think the h_ij should be calculated from the TP data and then perhaps use an assert to check it is indeed conformally flat.

Reply:
I'm happy to calculate the determinant explicitly but setting the conformal flatness (`vars.h[i][i] = 1.0`) explicitly is marginally more accurate so I will keep that. I can add a comment though.

Reply:
Ok sounds good.